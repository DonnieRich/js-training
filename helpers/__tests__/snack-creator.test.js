import { fs, vol } from 'memfs';
import { jest } from '@jest/globals';

jest.unstable_mockModule('fs/promises', () => (
    {
        default: {
            ...fs.promises
        }
    }
));

// console.log = jest.fn();
const { createScaffolding } = await import('../snack-creator');

describe('snackCreator', () => {

    beforeEach(() => {
        // vol.reset();
    });

    it('should work', async () => {

        vol.fromJSON(
            {
                [`${process.cwd()}/stubs/snack/snack.js`]: 'maremma',
                [`${process.cwd()}/stubs/snack/__tests__/snack.test.js`]: 'maremmaTest',
                [`${process.cwd()}/stubs/snack/.solution/snack.js`]: 'maremmaSolution',
                // [`${process.cwd()}/snacks/easy/no-problem/snack.js`]: 'ERR',
                [`${process.cwd()}/snacks/easy/nosweat.js`]: 'z'
            }
        );
        vol.mkdirSync(process.cwd(), { recursive: true });
        console.log(vol.toJSON())
        await createScaffolding('no-problem', 'easy');
        console.log(vol.toJSON())
        // expect(vol.toJSON()).toMatchSnapshot();
    })

})