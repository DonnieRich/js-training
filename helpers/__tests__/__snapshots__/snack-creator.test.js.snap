// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snackCreator should work 1`] = `
{
  "/WDA/Lavori/js-training/snacks/easy/no-problem/snack.js": "ERR",
  "/WDA/Lavori/js-training/snacks/easy/nosweat.js": "z",
  "/WDA/Lavori/js-training/stubs/snack/.solution/snack.js": "maremmaSolution",
  "/WDA/Lavori/js-training/stubs/snack/__tests__/snack.test.js": "maremmaTest",
  "/WDA/Lavori/js-training/stubs/snack/snack.js": "maremma",
}
`;
